app/livechat/server/lib/Livechat.js

```diff
@@ -307,11 +307,15 @@ export const Livechat = {
 		const updateUser = {
 			$set: {
 				token,
-				...(phone?.number ? { phone: [{ phoneNumber: phone.number }] } : {}),
+				...(phone && phone.number ? { phone: [{ phoneNumber: phone.number }] } : {}),
 				...(name ? { name } : {}),
 			},
 		};
 
+		if (phone === undefined) {
+			updateUser.$unset = { phone: [] };
+		}
+
```

Comment:
@AlexanderKanakis  @ear-dev I'm still seeing the same behavior. In the DB, the phone number is not stored.
<img width="1177" alt="Screenshot 2022-05-12 at 1 36 26 PM" src="https://user-images.githubusercontent.com/18264684/168023243-3b26fc20-331b-487c-86d5-f80f548aee47.png">

When I uncommented this code. It is being stored in the DB.

<img width="1177" alt="Screenshot 2022-05-12 at 1 38 22 PM" src="https://user-images.githubusercontent.com/18264684/168023331-0c0692b0-8eed-4631-b6cf-6167cef3b0bc.png">

And it needs to be stored in DB. Because... For Twilio integration, we use this phone number to send messages.

https://github.com/WideChat/Rocket.Chat/blob/develop_pwa/app/livechat/server/sendMessageBySMS.js#L63

@AlexanderKanakis I think the reason for this issue is because `registerGuest` function is used for registering users as well as updating users. So it can be called **multiple** times. And when registering/updating for some other property, phoneNumber becomes undefined and that unsets it in db